### ISSUE TITLE
[Android Studio support] Gradle build issue after creating.

### ISSUE THREAD

**Issue by iamchathu** [brodybits/create-react-native-module#120]
After creating new native android module, I opened android folder using Android Studio. It added missing gradle files and started the sync process and below error occurs.

```
INFO: API 'variant.getJavaCompile()' is obsolete and has been replaced with 'variant.getJavaCompileProvider()'.
It will be removed at the end of 2019.
```

After editing `android/build.gradle` to below it get fixed.

```gradle
  android.libraryVariants.all { variant ->
        def name = variant.name.capitalize()
        task "jar${name}"(type: Jar, dependsOn: variant.javaCompileProvider) {
            from variant.javaCompileProvider.get().destinationDir
        }
    }
```

**Comment by brodybits** [comment]
Does this mean that you tried to open a generated module with Android Studio?

**Comment by iamchathu** [comment]
@brodybits yes. I created Android only native module and open android folder from Android Studio and it added a gradle scripts and started to sync.

**Comment by brodybits** [comment]
Thanks. Marked as an enhancement as I am not yet motivated to support this case.

**Comment by brodybits** [comment]
The recommended approach is to generate the module with the example, then open the example project with Android Studio. You would likely have to copy artifacts between `example/node_modules` and `android` source tree if you develop in this manner.

Using Yarn workspaces (#128) *may* help us out.

Priority feature development with high-priority private support is available for commercial users, please contact <sales@xpbrew.consulting> for more details.

**Comment by brodybits** [comment]
This *may* be resolved by PR #135 in the 0.11.0 release.

**Comment by iamchathu** [comment]
@brodybits Any plans to make old version build projects to upgrade to new versions?

**Comment by brodybits** [comment]
> Any plans to make old version build projects to upgrade to new versions?

See #19, with workaround upgrade procedure now documented. I would appreciate if you would report whether or not this build issue is resolved by the upgrade.

I hope to make the upgrade more automatic, no promise of timeline. Priority is given to commercial customers, please contact <sales@xpbrew.consulting> for more information.

**Comment by chetstone** [comment]
This is unrelated to Android Studio. 
I see the same warning as @iamchathu and I don't use Android Studio.
I upgraded my module to RN 0.61 by creating a new module with this tool, then merging with the previous code. When I use my module in an app and compile with 
`-Pandroid.debug.obsoleteApi=true` I get this warning:

```
> Configure project :react-native-palette
WARNING: API 'variant.getJavaCompile()' is obsolete and has been replaced with 'variant.getJavaCompileProvider()'.
It will be removed at the end of 2019.
For more information, see https://d.android.com/r/tools/task-configuration-avoidance.
REASON: Called from: /Users/maint/prj/malas/upgrade/node_modules/react-native-palette/android/build.gradle:121
WARNING: Debugging obsolete API calls can take time during configuration. It's recommended to not keep it on at all times.
```

The offending [line 121 is here](https://github.com/chetstone/react-native-palette/blob/d239cd2a11636cef412e3838ecf7e51fbd11648b/android/build.gradle#L121). It is an exact copy of the line from the new library created with `create-react-native-module`

**Comment by brodybits** [comment]
Should be fixed now, I just merged PR #194 and published a new release.

In case of any further issues I would really appreciate a *new* issue with the standard [minimal, reproducible example ([1])](https://stackoverflow.com/help/minimal-reproducible-example).

[1] https://stackoverflow.com/help/minimal-reproducible-example
